gms/feature_service.cc

```diff
@@ -257,19 +257,24 @@ future<> persistent_feature_enabler::enable_features() {
     auto loaded_peer_features = co_await _sys_ks.load_peer_features();
     auto&& features = _g.get_supported_features(loaded_peer_features, gossiper::ignore_features_of_local_node::no);
 
+    auto features_v = boost::copy_range<std::set<std::string_view>>(features);
+    co_await _feat.enable(std::move(features_v), _sys_ks);
+}
+
+future<> feature_service::enable(std::set<std::string_view> features, db::system_keyspace& sys_ks) {
     // Persist enabled feature in the `system.scylla_local` table under the "enabled_features" key.
     // The key itself is maintained as an `unordered_set<string>` and serialized via `to_string`
     // function to preserve readability.
-    std::set<sstring> feats_set = co_await _sys_ks.load_local_enabled_features();
-    for (feature& f : _feat.registered_features() | boost::adaptors::map_values) {
+    std::set<sstring> feats_set = co_await sys_ks.load_local_enabled_features();
+    for (feature& f : registered_features() | boost::adaptors::map_values) {
         if (!f && features.contains(f.name())) {
             feats_set.emplace(f.name());
         }
     }
-    co_await _sys_ks.save_local_enabled_features(std::move(feats_set));
+    co_await sys_ks.save_local_enabled_features(std::move(feats_set));
 
     std::set<std::string_view> features_v = boost::copy_range<std::set<std::string_view>>(features);
-    co_await _feat.enable(std::move(features_v), feature_service::without_persisting{});
+    co_await enable(std::move(features_v), feature_service::without_persisting{});
```

Comment:
@xemul, thinking with modularity hat on:

feature_service should expose a feature_service_persistence_interface
main should implement feature_service_persistence_interface with system_keyspace
cql_test_env can implement feature_service_persistence_interface with std::unordered_set

Reply:
@avikivity ,  `gms::persistent_feature_enabler` can already implement feature_service_persistence_interface with system_keyspace , but it also needs gossiper. The question is where features-to-enable come from

Reply:
We can go wild and have every control-plane module declare its interfaces rather than talk to other control-plane modules directly.

However, without the build system enforcing isolation, we'll keep slipping and re-adding dependencies.

I'll send out something shortly.

Also cc @tchaikov who is relevant for modules, cmake, and modularity.